subsys/net/lib/lwm2m/lwm2m_obj_device.c

```diff
@@ -294,12 +370,25 @@ static int lwm2m_device_init(void)
 		LOG_DBG("Create LWM2M instance 0 error: %d", ret);
 	}
 
-	/* Create the default error code resource instance */
-	lwm2m_device_add_err(0);
+	/* Ensure error list is reset if not loaded from settings */
+	reset_error_list();
+
+	/* Load error code resource instances */
+	if (IS_ENABLED(CONFIG_LWM2M_DEVICE_ERROR_CODE_SETTINGS)) {
+		ret = settings_register(&lwm2m_obj_device_settings_handler);
+		if (ret != 0) {
+			LOG_ERR("Settings register failed: %d", ret);
+		}
+		ret = settings_load_subtree(SETTINGS_SUBTREE_LWM2M_OBJ_DEVICE);
```

Comment:
Shall we skip this in case `settings_register()` failed?

Reply:
Done